gradle-plugin: Update overly conservative input annotations

ting-yuan opened the pull request:
Classpath implies PathSensitive(PathSensitivity.RELATIVE)

gavra0 commented:
This looks good, but can you also add a test that:
- uses a custom location for the [build cache](https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_configure)
- creates a project, runs full build
- copies that project to another location and runs full build

In the second build, the ksp task should not run and it should be `from cache`.

ting-yuan commented:
@gavra0 I added a test case, which creates 2 projects that share a single build cache dir.

Just in case, I also removed `@get:InputFiles` per #625. There should be no difference besides [compatibility](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Classpath.html) to early Gradle versions (< 3.2). Since kapt also doesn't use `@InputFiles`, it should be safe?
